docker/client.py

```diff
@@ -274,18 +274,19 @@ def _warn_deprecated(self, arg_name, version):
     def _get_raw_response_socket(self, response):
         self._raise_for_status(response)
         if six.PY3:
-            sock = response.raw._fp.fp.raw._sock
+            sock = response.raw._fp.fp.raw
+            sock._response = response
         else:
             sock = response.raw._fp.fp._sock
-        try:
-            # Keep a reference to the response to stop it being garbage
-            # collected. If the response is garbage collected, it will close
-            # TLS sockets.
-            sock._response = response
-        except AttributeError:
-            # UNIX sockets can't have attributes set on them, but that's fine
-            # because we won't be doing TLS over them
-            pass
```

Comment:
Can you expand on those changes here? It seems to me that both versions would do exactly the same thing, am I mistaken?